### ISSUE TITLE
The initial DAREBEE_URL is downloaded twice

### ISSUE THREAD

**Issue by eumiro** [jugmac00/workout-bot#3]
DAREBEE_URL should be downloaded once only and then used in both `extract_*` methods.

**Comment by jugmac00** [comment]
Good catch!

This is certainly true. Also, it would make sense to pass in the html into both the functions.

Then, the HTML would be downloaded only once, and both the `extract`-methods would be trivially to test (text in, text out, no side effects).

I will delay this refactoring, as I plan to move the `extract`-methods (or let's call them image-URL providers or workouts) to another module.

The plan is to make both the images (ie workouts), and maybe the medium (for now only Telegram) configurable for the user.

Currently, I am looking for a way how to wire up workouts/transport medium in a user friendly way.

To wrap this up - you are certainly right. Nevertheless, I will postpone the refactoring a bit. And I will leave a comment at both functions, so it is also clear for other readers of the code.
